mkdocs_awesome_pages_plugin/navigation.py

```diff
@@ -206,29 +221,48 @@ def __init__(
         explicit_sections: Set[Section],
     ):
         self.options = options
-        self.sections = {}
+        self.sections: Dict[Section, Meta] = {}
         self.docs_dir = docs_dir
         self.explicit_sections = explicit_sections
 
-        root_path = self._gather_metadata(items)
+        root_path, root_pages = self._gather_metadata(items)
         self.root = Meta.try_load_from(join_paths(root_path, self.options.filename))
 
-    def _gather_metadata(self, items: List[NavigationItem]) -> Optional[str]:
-        paths = []
+        if self.root.order_by == Meta.ORDER_BY_TITLE:
+            for page in root_pages:
+                assure_title_for_page(page)
+
+    def _gather_metadata(self, items: List[NavigationItem]) -> Optional[Tuple[str, List[Page]]]:
+        paths: List[str] = []
+        pages: List[Page] = []
```

Comment:
I find this `pages` variable (and return value) a bit unclear because as far as I can tell it's only used when ordering by title.

Reply:
I added what was needed for the title ordering based on local `.pages` settings. 
Once again I focused on avoiding redoing the same task again, in this case looping over the whole list of Page items, in another part of the code.
I guess I could connect the 2 lists into a `Dict[str, Optional[Page]]`, in `_common_dirname` loop over the `keys` and in `_gather_metadata` loop over filtered `values` (sections don't have Pages and are `None`), but I'm not sure if that would be cleaner 🤔.
Another less performant way, but I guess cleaner in code, would be adding some intersecting function, which would loop over the Page `values` and return the `_common_dirname` results 🤔